core/client/src/main/java/alluxio/client/block/StreamFactory.java

```diff
@@ -43,6 +44,10 @@ private StreamFactory() {} // prevent instantiation
    */
   public static BlockOutStream createLocalBlockOutStream(FileSystemContext context, long blockId,
       long blockSize, WorkerNetAddress address, OutStreamOptions options) {
+    if (NettyClient.isDomainSocketSupported(address)) {
+      return BlockOutStream
+          .createNettyBlockOutStream(blockId, blockSize, address, context, options);
+    }
```

Comment:
I've renamed "local" to "short circuit" in Block{In,Out}Stream but left the names in StreamFactory unchanged. Let me know that is not what you want.